net/ipv4/ip_sockglue.c

```diff
@@ -604,6 +606,19 @@ static int do_ip_setsockopt(struct sock *sk, int level,
 			inet->tos = val;
 			sk->sk_priority = rt_tos2priority(val);
 			sk_dst_reset(sk);
+			/* Update TOS on mptcp subflow */
+			if (is_meta_sk(sk)) {
+				struct sock *sk_it, *tmpsk;
+				mptcp_for_each_sk_safe(tcp_sk(sk)->mpcb, sk_it, tmpsk) {
+					if (tcp_sk(sk_it)->closing || tcp_sk(sk_it)->send_mp_fclose)
```

Comment:
Is this here really needed? I think it is safe to set the TOS-bits even on subflows that are about to get closed, no?